src/PrestaShopBundle/Bridge/Smarty/ToolbarFlagsHydrator.php

```diff
@@ -0,0 +1,180 @@
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShopBundle\Bridge\Smarty;
+
+use Configuration;
+use Cookie;
+use Language;
+use PrestaShop\PrestaShop\Adapter\LegacyContext;
+use PrestaShopBundle\Bridge\Controller\ControllerConfiguration;
+use PrestaShopBundle\Bridge\Utils\CookieFilterUtils;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+use Tools;
+
+/**
+ * Class ToolbarFlagsHydrator hydrate toolbar flags in the Controller configuration
+ */
+class ToolbarFlagsHydrator implements HydratorInterface
+{
+    /**
+     * @var Cookie
+     */
+    private $cookie;
+
+    /**
+     * @var RouterInterface
+     */
+    private $router;
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    public function __construct(RouterInterface $router, TranslatorInterface $translator, LegacyContext $legacyContext)
+    {
+        $this->router = $router;
+        $this->translator = $translator;
+        $this->cookie = $legacyContext->getContext()->cookie;
+    }
+
+    /**
+     * @param ControllerConfiguration $controllerConfiguration
+     *
+     * @return void
+     */
+    public function hydrate(ControllerConfiguration $controllerConfiguration)
+    {
+        $this->initToolbar($controllerConfiguration);
+        $this->initPageHeaderToolbar($controllerConfiguration);
+
+        $controllerConfiguration->templatesVars['maintenance_mode'] = !(bool) Configuration::get('PS_SHOP_ENABLE');
+        $controllerConfiguration->templatesVars['debug_mode'] = (bool) _PS_MODE_DEV_;
+        $controllerConfiguration->templatesVars['lite_display'] = $controllerConfiguration->liteDisplay;
+        $controllerConfiguration->templatesVars['url_post'] = $this->router->generate('admin_features_index');
+        $controllerConfiguration->templatesVars['show_page_header_toolbar'] = $controllerConfiguration->showPageHeaderToolbar;
+        $controllerConfiguration->templatesVars['page_header_toolbar_title'] = $controllerConfiguration->pageHeaderToolbarTitle;
+        $controllerConfiguration->templatesVars['title'] = $controllerConfiguration->pageHeaderToolbarTitle;
+        $controllerConfiguration->templatesVars['toolbar_btn'] = $controllerConfiguration->pageHeaderToolbarButton;
+        $controllerConfiguration->templatesVars['page_header_toolbar_btn'] = $controllerConfiguration->pageHeaderToolbarButton;
+        $controllerConfiguration->templatesVars['help_link'] = 'https://help.prestashop.com/' . Language::getIsoById($controllerConfiguration->user->getData()->id_lang) . '/doc/'
```

Comment:
`@todo` to remember to separate this URL for `const` and replace it because of the split?